Review followup: PR #23667 — GH#23649: exercise worktree owner registration

## Unaddressed review bot suggestions

PR #23667 was merged with unaddressed review bot feedback. Each comment
below includes its file path, line number, a direct link to the inline
review comment, and a `diff` fence with the code context the bot was
flagging. Resolved and outdated threads are filtered out via GitHub's
GraphQL review-thread state. Read the relevant lines, decide whether
the suggestion is correct, and either apply the fix or close this issue
with a wontfix rationale.

**Source PR:** https://github.com/marcusquinn/aidevops/pull/23667

---

### You are the triager (worker-is-triager rule)

This issue is **auto-created from review bot output** and dispatched
directly to you. Review bots can be wrong: hallucinated line refs, false
premises about codebase structure, template-driven sweeps without
measurements (see GH#17832-17835 for prior art and `AGENTS.md`
"AI-Generated Issue Quality"). **Do not assume the bot is correct.** Verify before acting.

You must end in exactly one of three outcomes — no fourth "hand it back
to the human" path exists. Humans approve decisions; they do not re-do
analysis.

#### Outcome A — Premise falsified → close the issue

1. Read the cited `file:line` (listed under *Files to modify* below).
2. If the bot's claim is factually wrong (file doesn't exist at that
   line, function doesn't behave as described, "auto-generated" section
   isn't actually auto-generated, etc.), **close the issue** with a
   comment in this shape:

   > **Premise falsified.** \<what the bot claimed\>. \<what the code
   > actually shows, with a `file:line` citation or one-line quote\>.
   > Not acting.

   No PR. No further dispatch. The closing comment trains the next
   session reading this thread and the noise filter.

#### Outcome B — Premise correct + fix is obvious → implement and PR

1. Verify the bot's premise as above.
2. Read the Worker Guidance section below, open a worktree, implement.
3. Open a PR with `Resolves #<this-issue-number>` in the body
   (use THIS issue's number, not the source PR's) so merge auto-closes it.
4. Follow the normal Lifecycle Gate (brief, tests, review-bot-gate,
   merge, postflight).

#### Outcome C — Premise correct but approach is a genuine judgment call

Only use this path if you reach it after Outcomes A and B don't apply:
the bot's finding is real, but the fix requires a decision that is
architectural, policy, breaking-change, or otherwise genuinely outside
what you can resolve autonomously. In that case, post a **decision
comment** with exactly these fields:

- **Premise check:** one line, confirming the finding is real.
- **Analysis:** 2-4 bullets on the trade-offs.
- **Recommended path:** the option you would take if the decision were
  yours, with rationale.
- **Specific question:** the single decision the human needs to make
  (yes/no or pick-one, not open-ended).

Then apply `needs-maintainer-review` and stop. The human wakes up to a
ready-to-approve recommendation, not a blank task.

> **Ambiguity about scope or style is not Outcome C.** Per
> `AGENTS.md` "Reasoning responsibility", the model does the
> thinking and delivers a recommendation. Only escalate what is genuinely
> a maintainer-only decision.

### Worker Guidance

**Files to modify:**

- `.agents/scripts/tests/test-dispatch-single-issue-helper.sh:121`
- `.agents/scripts/tests/test-dispatch-single-issue-helper.sh:210`
- `.agents/scripts/tests/test-dispatch-single-issue-helper.sh:646`

**Implementation steps (Outcome B path):**

1. Read the `diff` block under each inline comment below — it shows the
   exact code the bot was flagging. Open the file only if you need
   surrounding context beyond what the diff tail shows.
2. Read the bot's full comment below the diff — it contains the rationale
   and any suggested change.
3. Verify the premise before implementing (see Outcome A). If the premise
   is wrong, switch to Outcome A instead of burning iterations trying to
   satisfy a wrong suggestion.
4. If multiple comments target the same file, group your edits into one
   logical commit.
5. Run `shellcheck` / `markdownlint-cli2` / project tests as appropriate.

**Verification:**

- Open the new PR with `Resolves #<this-issue>` so this followup is auto-closed on merge.
- If the bot's suggestion was incorrect, close this issue with a Outcome A comment — do not open a no-op PR.

### Inline comments

#### gemini-code-assist on `.agents/scripts/tests/test-dispatch-single-issue-helper.sh:121`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/23667#discussion_r3250379538)

```diff
@@ -107,6 +114,41 @@ _install_mock_set_origin_label() {
 	return 0
 }
 
+# shellcheck disable=SC2317
+register_worktree() {
+	printf 'register_worktree %s\n' "$*" >>"$REGISTER_WORKTREE_LOG"
+	return 0
+}
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> The `register_worktree` mock will cause the script to exit with an error if `REGISTER_WORKTREE_LOG` is empty (its initial state), because `>> ""` is an invalid redirection in Bash. Since `set -e` is active at the top of the script, this will terminate the entire test suite if `register_worktree` is called by any code path that hasn't explicitly configured the log path. It is safer to check if the variable is non-empty before attempting to write to it. Additionally, ensure the function propagates the exit code of the commands it runs instead of masking them with a hardcoded `return 0`.
> 
> ```suggestion
> register_worktree() {
> 	if [[ -n "${REGISTER_WORKTREE_LOG:-}" ]]; then
> 		printf 'register_worktree %s\n' "$*" >>"$REGISTER_WORKTREE_LOG"
> 		return $?
> 	fi
> }
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. In shell scripts, add guard clauses to check if a variable is empty before using it, especially when its value is derived from an external command that could fail or produce no output.
> 2. In shell scripts, functions that wrap a command should propagate its exit code to the caller. Avoid masking potential errors with a hardcoded return 0. Instead, use return $? or capture the exit code and return it.
> </details>

#### gemini-code-assist on `.agents/scripts/tests/test-dispatch-single-issue-helper.sh:210`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/23667#discussion_r3250379561)

```diff
 }
 
+# shellcheck disable=SC2317
+_dsi_repo_path_for_slug() {
+	local repo_slug="$1"
+	if [[ "$repo_slug" == "owner/repo" && -n "$MOCK_REPO_PATH" ]]; then
+		printf '%s\n' "$MOCK_REPO_PATH"
+		return 0
+	fi
+	printf '\n'
+	return 1
+}
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> The mock function returns a literal newline character when the repository slug does not match. This can lead to unexpected behavior in the caller if it checks for a non-empty string (e.g., `[[ -n "$path" ]]`), as a newline is considered non-empty. It is better to return nothing and rely on the exit code or an empty string. Additionally, include a guard clause to verify that `MOCK_REPO_PATH` is set before use.
> 
> ```suggestion
> _dsi_repo_path_for_slug() {
> 	local repo_slug="$1"
> 	if [[ "$repo_slug" == "owner/repo" && -n "$MOCK_REPO_PATH" ]]; then
> 		printf '%s\n' "$MOCK_REPO_PATH"
> 		return $?
> 	fi
> 	return 1
> }
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. In shell scripts, add guard clauses to check if a variable is empty before using it, especially when its value is derived from an external command.
> </details>

#### gemini-code-assist on `.agents/scripts/tests/test-dispatch-single-issue-helper.sh:646`

[View inline comment](https://github.com/marcusquinn/aidevops/pull/23667#discussion_r3250379570)

```diff
+		"$helper_call" == *"--issue 12345"* ]]; then
+		passed=0
 	fi
-	print_result "worktree creation registers dispatch owner metadata" "$failed"
+
+	_DSI_WORKTREE_HELPER="$original_worktree_helper"
+	MOCK_GIT_WORKTREE_LIST="0"
+	MOCK_REPO_PATH=""
+	MOCK_WORKTREE_PATH=""
+	MOCK_WORKTREE_BRANCH=""
+	MOCK_DATE_UTC=""
+	rm -rf "$test_dir"
```

> ![medium](https://www.gstatic.com/codereviewagent/medium-priority.svg)
> 
> The test function modifies global log path variables and exports `MOCK_WORKTREE_HELPER_LOG`, but it does not reset or unset them during cleanup. This can cause side effects in subsequent tests. Additionally, ensure that temporary files are removed as soon as they are no longer needed rather than waiting until the end of the function.
> 
> ```suggestion
> _DSI_WORKTREE_HELPER="$original_worktree_helper"
> 	MOCK_GIT_WORKTREE_LIST="0"
> 	MOCK_REPO_PATH=""
> 	MOCK_WORKTREE_PATH=""
> 	MOCK_WORKTREE_BRANCH=""
> 	MOCK_DATE_UTC=""
> 	REGISTER_WORKTREE_LOG=""
> 	unset MOCK_WORKTREE_HELPER_LOG
> 	MOCK_WORKTREE_HELPER_LOG=""
> 	rm -rf "$test_dir"
> ```
> 
> <details>
> <summary>References</summary>
> 
> 1. In shell scripts, remove temporary files as soon as their content has been read and they are no longer needed, rather than waiting until the end of the function.
> </details>

### PR review summaries

#### gemini-code-assist[bot] review summary

[View review](https://github.com/marcusquinn/aidevops/pull/23667#pullrequestreview-4300671517)

> ## Code Review
> 
> This pull request enhances the test suite for the dispatch helper script by introducing mocks for `git`, `date`, and `register_worktree`, and refactoring the worktree registration test to use a temporary directory for more isolated verification. Feedback was provided to prevent a potential script crash caused by invalid redirection in the `register_worktree` mock, to ensure `_dsi_repo_path_for_slug` returns an empty string instead of a newline on failure to avoid false-positive shell checks, and to properly reset or unset global mock variables during test cleanup to prevent side effects in subsequent tests.
<!-- aidevops:sig -->
---
[aidevops.sh](https://aidevops.sh) v3.15.54 automated scan.

--- a/.agents/scripts/tests/test-dispatch-single-issue-helper.sh
+++ b/.agents/scripts/tests/test-dispatch-single-issue-helper.sh
@@ -116,8 +116,11 @@ _install_mock_set_origin_label() {
 
 # shellcheck disable=SC2317
 register_worktree() {
+	if [[ -z "${REGISTER_WORKTREE_LOG:-}" ]]; then
+		return 0
+	fi
 	printf 'register_worktree %s\n' "$*" >>"$REGISTER_WORKTREE_LOG"
-	return 0
+	return $?
 }
 
 # shellcheck disable=SC2317
@@ -205,7 +208,6 @@ _dsi_repo_path_for_slug() {
 		printf '%s\n' "$MOCK_REPO_PATH"
 		return 0
 	fi
-	printf '\n'
 	return 1
 }
 
@@ -630,6 +632,7 @@ test_create_worktree_registers_dispatch_owner() {
 	registered=$(<"$REGISTER_WORKTREE_LOG")
 	local helper_call=""
 	helper_call=$(<"$MOCK_WORKTREE_HELPER_LOG")
+	rm -f "$REGISTER_WORKTREE_LOG" "$MOCK_WORKTREE_HELPER_LOG"
 	local passed=1
 	if [[ "$rc" -eq 0 && "$registered" == *"register_worktree ${MOCK_WORKTREE_PATH} ${MOCK_WORKTREE_BRANCH}"* &&
 		"$registered" == *"--task 12345"* && "$registered" == *"--session dispatch-precreate-12345"* &&
@@ -643,6 +646,9 @@ test_create_worktree_registers_dispatch_owner() {
 	MOCK_WORKTREE_PATH=""
 	MOCK_WORKTREE_BRANCH=""
 	MOCK_DATE_UTC=""
+	REGISTER_WORKTREE_LOG=""
+	unset MOCK_WORKTREE_HELPER_LOG
+	MOCK_WORKTREE_HELPER_LOG=""
 	rm -rf "$test_dir"
 	print_result "worktree creation registers dispatch owner metadata" "$passed" "rc=$rc registered=$registered helper=$helper_call"
 	return 0

